openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
+        var temp = {};
+        temp.region = la[ia].properties[selectedRegion];
+        indicator.push(temp);
+    }
+    for (var i = 0; i < index.length; i++) {
+        for (var j = 0; j < la.length; j++, ct++) {
+            if (indicator[j].region == la[j].properties[selectedRegion]) {
+                var tempName = index[i].name;
+                var tempValue = la[j].properties[tempName];
+                indicator[j][tempName] = tempValue;
             }
         }
-
-        // Get unique category names
-        var categoryUnique = categoryList.filter(function(itm,i,categoryList){
-            return i==categoryList.indexOf(itm);
-        });
-
-        for (var l in categoryUnique) {
-            // Append category names to dropdown list
-            var categoryTitle = categoryUnique[l];
-            var opt = document.createElement('option');
-            opt.innerHTML = categoryTitle;
-            opt.value = categoryTitle;
-            selCat.appendChild(opt);
-            // Append layer list to dowpdown
-            var layerOpt = document.createElement('option');
+    }
+    return indicator;
+}
+
+function combineIndicators(nameLookUp, themeObj, JSONthemes) {
+    projectDef = sessionProjectDef;
+    var ct = 0;
+    var subIndex = {};
+    var operator;
+    for (var y = 0; y < projectDef.children.length; y++) {
+        if (projectDef.children[y].name == nameLookUp) {
+            operator = projectDef.children[y].operator;
         }
-
-    });
-
-    // Create dynamic categorized layer dialog
-    $('#layer-category').change(function() {
-        // Remove the layer list element
-        document.getElementById('layer-list').options.length = 0;
-
-       // Create the layer list ibased on the category selected
-        var e = document.getElementById('layer-category');
-        var strUser = e.options[e.selectedIndex].value;
-        var layersArray = layersByCat[strUser];
-
-        for (var i in layersArray) {
-            var layers = layersArray[i];
-            var opt = document.createElement('option');
-            opt.innerHTML = layers;
-            opt.valuse = layers;
-            selLayer.appendChild(opt);
+    }
+    // first create an object with all of the district names
+    for (var t = 0; t < themeObj.length; t++) {
+        var tempRegion = themeObj[t].region;
+        subIndex[tempRegion] = 0;
+    }
+    // get some info aobut the themes
+    var themekeys = [];
+    var themeWeightObj = {};
+    for (var u = 0; u < JSONthemes.length; u++) {
+        var themeName = JSONthemes[u].name;
+        var themeWeight = JSONthemes[u].weight;
+        themekeys.push(themeName);
+        themeWeightObj[themeName] = themeWeight;
+    }
+    // compute the subIndex values based on operator
+    if (operator == 'Simple sum (ignore weights)') {
+        for (var v = 0; v < themeObj.length; v++) {
+            var tempElementValue = 0;
+            var themeObjMunic = themeObj[v].region;
```

Comment:
In the past I had a discussion with Chris about what name we should use. We agreed that "zone" is a good choice because it is generic enough to avoid overlapping with other possible concepts. For instance, "region" could be confused with some sub-national zones, or with some multi-national named-regions. And "zone" is a word that is not unknown in the geographical context; for instance in QGIS there are some algorithms called something like "zonal statistics". For those reasons, after a thorough discussion with Chris, we decided to refactor all the code in the plugin to use "zone" instead of other possible candidates that we were using interchangeably. I believe it would be good to use a common naming convention in our tools. Nevertheless, if you feel more comfortable with the word "region", I am not strongly against it. Whatever you decide, just make sure you use one single convention in the whole web viewer, avoiding confusion with districts, municipalities, etc.